core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -20,11 +20,17 @@
 
 import org.apache.iceberg.exceptions.CommitFailedException;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.view.ViewMetadata;
 
 /** Represents a requirement for a {@link MetadataUpdate} */
 public interface UpdateRequirement {
   void validate(TableMetadata base);
 
+  default void validate(ViewMetadata base) {
+    throw new UnsupportedOperationException(
```

Comment:
Should we use a `ValidationException` instead since using an assertion on a view in this case is invalid?

Reply:
makes sense to use `ValidationException` here, I've done that

Reply:
@nastra, looks like this didn't get updated.

Reply:
my bad, I forgot to push those changes